backend/meson.build

```diff
@@ -1,11 +1,13 @@
 wlr_files += files('backend.c')
 
-subdir('drm')
-subdir('headless')
-subdir('libinput')
+backends = get_option('backends')
+
+foreach backend : ['drm', 'noop', 'headless', 'libinput', 'x11', 'wayland']
+  if [backend, 'auto'].contains(backends)
```

Comment:
This looks like it will evaluate to true if `backends` in its entirety is equal to `backend` or to "auto". I imagine you wanted it to be true if one of the *elements* in `backends` was equal to `backend` or "auto". Am I missing some magic here?